container-search/src/main/java/com/yahoo/prelude/query/CompositeItem.java

```diff
@@ -221,33 +221,47 @@ public CompositeItem clone() {
     }
 
     private void fixConnexity(CompositeItem copy) {
-        List<Item> flatland = new ArrayList<>();
-        List<Item> flatCopy = new ArrayList<>();
-        taggingFlatten(this, flatland);
+        int hasUniqueIdCount = countUniqueId(this);
+        if (hasUniqueIdCount == 0) return;
+        List<Item> flat = new ArrayList<>(hasUniqueIdCount);
+        List<Item> flatCopy = new ArrayList<>(hasUniqueIdCount);
+        taggingFlatten(this, flat);
         taggingFlatten(copy, flatCopy);
-        int barrier = flatland.size();
+        int barrier = flat.size();
         for (int i = 0; i < barrier; ++i) {
-            Item orig = flatland.get(i);
-            int connectedTo = find(orig.connectedItem, flatland);
+            Item orig = flat.get(i);
+            int connectedTo = find(orig.connectedItem, flat);
             if (connectedTo >= 0) {
                 TaggableItem tagged = (TaggableItem) flatCopy.get(i);
                 tagged.setConnectivity(flatCopy.get(connectedTo), orig.connectivity);
             }
         }
     }
 
-    private void taggingFlatten(Item tree, List<Item> container) {
+    private static int countUniqueId(Item tree) {
```

Comment:
This assumes that no item that has an unique id can have any children that has an unique id. Is that guaranteed by anything? I couldn't find it.

Reply:
Good point. I  assumed that the logic in taggingFlatten was correct.
If not this is an issue in current code too.
I will see what happens if I change it.